src/vanilla/ClientModelExtensions.cs

```diff
@@ -198,10 +198,10 @@ public static string GetFormattedReferenceValue(this Parameter parameter)
                 primaryType = New<PrimaryType>(KnownPrimaryType.String);
             }
 
-            if (primaryType == null || primaryType.KnownPrimaryType != KnownPrimaryType.String)
+            if (primaryType == null)
             {
                 throw new InvalidOperationException(
-                    "Cannot generate a formatted sequence from a " + $"non-string array parameter {parameter}");
+                    "Cannot generate a formatted sequence from a " + $"null parameter {parameter}");

```

Comment:
please update the comment to the function if any primary type is supported.

Reply:
Updated the comment